File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 12/6/2025, 5:55:29 am
class Solution {
public:
    int maxArea(vector<int>& height) {
        unsigned int ret = 0;
        //unsigned int tmp;
        for (unsigned int i=0; i<height.size()-1; i++){
            for (unsigned int j=i+1; j<height.size(); j++){
                unsigned tmp = height[i];
                if (height[i] > height[j]) tmp = height[j];
                if (tmp * ((j+1)-(i+1)) > ret){
                    ret = tmp * ((j+1)-(i+1));
                }
            }
        }return ret;
    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 4/10/2025, 5:17:58 pm
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)

        best = 0
        l = 0
        r = n-1

        while l < r:
            area = min(height[l], height[r]) * (r - l)
            best = max(area, best)

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        
        return best